### PR TITLE
[aievec] to-llvm flow for aievec.ext op

### PR DESCRIPTION
**Pull request by jamestcl-amd**
This PR add the support for aievec.ext op going through the to-llvm flow.

Changes:
- Add ext intrinsic ops to XLLVM dialect.
- Add aievec-to-llvm conversion pattern/tests for the aievec.ext op.
- Add test for translating to external llvm.

### CODE REVIEW

File: test/Target/LLVMIR/aievec.mlir [hunk 1]
```diff
@@ -222,6 +222,33 @@ llvm.func @ext_i256_i512(%v : vector<16xi32>, %idx : i32) -> vector<8xi32> {
...
+    // CHECK: call <8 x i32> @llvm.aie2.ext.I256.I1024(
+    // CHECK-SAME: <32 x i32> %{{[0-9]+}}, i32 %{{[0-9]+}})
+    %1 = "xllvm.intr.aie2.ext.I256.I1024"(%v, %idx) : 
+                                        (vector<32xi32>, i32) -> vector<8xi32>
+    llvm.return %1 : vector<8xi32>
+}
+
+// CHECK-LABEL: define <4 x i32> @ext_i128_i512
+llvm.func @ext_i128_i512(%v : vector<16xi32>, %idx : i32) -> vector<4xi32> {
+    // CHECK: call <4 x i32> @llvm.aie2.ext.I128.I512(
```
Comment (jsetoain) [inline test/Target/LLVMIR/aievec.mlir:245]: It should say _"@llvm.aie2.extract.I128.I512"_

**Review by jsetoain — APPROVED** [review]
Other than the typo in the `CHECK` line, everything looks good.

File: include/aie/Dialect/XLLVM/IR/XLLVMAIE2IntrOps.td [hunk 2]
```diff
@@ -185,6 +185,24 @@ def ExtI256I512IntrOp :
...
+                   I32:$idx)>;
+
+def ExtI256I1024IntrOp :
+    AIEVec2_IntrOp<"ext.I256.I1024",
+        [TypeIs<"res", VectorOfLengthAndType<[8], [I32]>>]>,
+    Arguments<(ins VectorOfLengthAndType<[32], [I32]>:$src,
+                   I32:$idx)>;
+
+def ExtI128I512IntrOp :
+    AIEVec2_IntrOp<"extract.I128.I512",
```
Comment (jsetoain) [inline include/aie/Dialect/XLLVM/IR/XLLVMAIE2IntrOps.td:201]: Alternatively, rename this here to `ext.I128.I512`.

Comment (jamestcl-amd) [reply include/aie/Dialect/XLLVM/IR/XLLVMAIE2IntrOps.td:201]: The external llvm intrinsic is `def int_aie2_extract_I128_I512 : ClangBuiltin<"__builtin_aiev2_extract_I128_I512">, AIEV2Extract_I128_I512;`, so it needs to be `"extract.I128.I512"`.

**Pull request merged**